### PR TITLE
Include ssh gateway public key in UpdateTunnelEndpoint

### PR DESCRIPTION
**Pull request by klvnraju**
Send additional parameter to include ssh gateway public key when calling UpdateTunnelEndpoint api if there is atleast one ssh port

### CODE REVIEW

File: cs/src/Management/TunnelManagementClient.cs [hunk 1]
```diff
@@ -921,13 +921,19 @@ public void Dispose()
             Requires.NotNull(endpoint, nameof(endpoint));
             Requires.NotNullOrEmpty(endpoint.HostId!, nameof(TunnelEndpoint.HostId));
 
+            string? query = null;
+            if (tunnel.Ports != null && tunnel.Ports.Where((p) => p.Protocol == TunnelProtocol.Ssh).Any())
+            {
+                query = "includeSshGatewayPublicKey=true";
```
Comment (IlyaBiryukov) [inline cs/src/Management/TunnelManagementClient.cs:927]: Can `includeSshGatewayPublicKey` be a string constant in a contract? Then tunnel service may also use that constant.

**Review by IlyaBiryukov — DISMISSED** [review]

File: cs/src/Management/TunnelManagementClient.cs [hunk 2]
```diff
@@ -921,13 +921,19 @@ public void Dispose()
             Requires.NotNull(endpoint, nameof(endpoint));
             Requires.NotNullOrEmpty(endpoint.HostId!, nameof(TunnelEndpoint.HostId));
 
+            string? query = null;
+            if (tunnel.Ports != null && tunnel.Ports.Where((p) => p.Protocol == TunnelProtocol.Ssh).Any())
```
Comment (jasongin) [inline cs/src/Management/TunnelManagementClient.cs:925]: nit: `.Any(lambda)` instead of `.Where(lambda).Any()`

**Comment by jasongin** [comment]
An alternative approach could be to add a `TunnelRequestOptions.AdditionalQueryParameters`, similar to `AdditionalHeaders`.

**Comment by jasongin** [comment]
Why is the option only on update operation and not get operation? Can you explain the scenario where this is used?

**Comment by klvnraju** [comment]
There is no get for endpoints api.
The tunnel CLI uses this update api while hosting a tunnel and uses the output to show relay url/ssh command.

File: cs/src/Management/TunnelManagementClient.cs [hunk 3]
```diff
@@ -921,13 +921,19 @@ public void Dispose()
...
+            {
+                query = "includeSshGatewayPublicKey=true";
+            }
```
Comment (klvnraju) [inline cs/src/Management/TunnelManagementClient.cs:928]: Do this in typescript SDK as well

**Comment by jasongin** [comment]
> There is no get for endpoints api.

The get-tunnel response includes endpoints.

**Comment by klvnraju** [comment]
The get-tunnel implementation has port information to determine if there is an ssh port and hence does not require this query parameter. (Only exception is if we call get-tunnel with a non-ssh port number and there are other ports which are ssh.. but in that scenario, we can skip sending the key)

**Pull request closed without merging**